### PR TITLE
updated

### PR DESCRIPTION
**Pull request by JamesB05**

### CODE REVIEW

**Comment by Chlo3russell** [comment]
1. Have the requirements been met?

The requirements have been met but in your own way, the game class is done seperately to the "roller" method. So you have created an additional class then specified in the brief. The Debug.Assert is used and called correctly.  Overall the brief has been met with the additional functionality.


2. Is the code formatted using the Style Guidelines correctly?

Comparing the format to the guidelines provided (https://google.github.io/styleguide/csharp-style.html) they are now used correcly there are some instances of variable that aren't named appropriately to there protection level. The organisation could use some improvement. Consider reading the correct organistion on the document as well.

3. Is the code easy to read?

The code is easy to read once you have already read through it. The comments could use a little more detail, you could point out what the code is aiming to do (such as object instantiation and encapsulation).  The XML Comments could be more frequent and the returns need to be below the summary!

4. Are different errors handled correctly?

Debug.Assert has been used and should handle and erroneous results, I didn't see anything around the while loop. Could there be something added here if the user doesn't write "yes" or "no"?

Overall it works and shows good basic knowledge of C#!

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ internal class Die
         private static Random random = new Random();
```
Comment (jozefbonnar) [inline CMP1903_A1_2324/Die.cs:19]: The variable name `_Faces` should be changed to `_faces` to comply with the style guidelines on line 17 and 26.

File: CMP1903_A1_2324/Die.cs
Comment (jozefbonnar) [inline CMP1903_A1_2324/Die.cs:1]: The variable names `Rolls` and `SumOfRolls` need to be changed to adhere to style guidelines as they are private:
```c#
    private List<int> _rolls = new List<int>();
    private int _sumOfrolls = 0;

Comment (jozefbonnar) [reply CMP1903_A1_2324/Die.cs:1]: The variable `_valueOfRoll` needs to be changed to `valueOfRoll` in the method `DiceRollOnce` to adhere to the style guidelines.
```c#
        public int DiceRollOnce()
        {
            Die die = new Die();
            int valueOfRoll = die.Roll();
            _rolls.Add(valueOfRoll);
            _sumOfRolls += valueOfRoll;  // Update the sum with each roll
            return valueOfRoll;  /// <returns>The value rolled on the die.</returns>
        }

Comment (jozefbonnar) [reply CMP1903_A1_2324/Die.cs:1]: In the method `ContinuousDieRoller` the variable `_rollValue` needs to be changed to `rollValue` as it's not private.

File: CMP1903_A1_2324/Testing.cs [hunk 2]
```diff
@@ -18,22 +18,22 @@ internal class Testing
...
-            Debug.Assert(_rollValue2 >= 1 && _rollValue2 <= 6, "Die 2 is out of range.");
-            Debug.Assert(_rollValue3 >= 1 && _rollValue3 <= 6, "Die 3 is out of range.");
+            Debug.Assert(rollValue1 >= 1 && rollValue1 <= 6, "Die 1 is out of range.");
+            Debug.Assert(rollValue2 >= 1 && rollValue2 <= 6, "Die 2 is out of range.");
+            Debug.Assert(rollValue3 >= 1 && rollValue3 <= 6, "Die 3 is out of range.");
 
             // Confirms that the sum of die rolls is as expected and if sum is not as expected it produces an error message
-            int _sumVerified = _rollValue1 + _rollValue2 + _rollValue3;
+            int _sumVerified = rollValue1 + rollValue2 + rollValue3;
             Debug.Assert(_sumVerified == _sumVerified, $"The value of sum is not as expected. Expected value: {_sumVerified}, Actual value: {_sumVerified}");
```
Comment (jozefbonnar) [inline CMP1903_A1_2324/Testing.cs:37]: Changing variable to adhere to style guidelines.
```suggestion
            int sumVerified = rollValue1 + rollValue2 + rollValue3;
            Debug.Assert(sumVerified == sumVerified, $"The value of sum is not as expected. Expected value: {sumVerified}, Actual value: {sumVerified}");
```

File: CMP1903_A1_2324/Program.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ static void Main(string[] args)
                 {
                     int _rollValue = _diceRoller.DiceRollOnce();
```
Comment (jozefbonnar) [inline CMP1903_A1_2324/Program.cs:26]: The variables `_diceRoller` and `_rollValue` needs to be updated to `diceRoller` and `rollValue`.
```suggestion
                    int rollValue = diceRoller.DiceRollOnce();
```

**Review by jozefbonnar — COMMENTED** [review]
## Summary

### Requirements Check:

**Have the requirements been met?** 

- Yes, the requirements have been met and exeeded it, as the code implements four classes as requested: Die, Game, and Testing then an additional DiceRoller class.

- 1. The Die class contains a property to hold the current die value and a method for rolling the die.
- 2. The Game class sums the values of the 3 die rolls.
- 3. The Testing class test all 3 of the dice using `Debug.Assert()` and also verifies the sum.
- 4. The DiceRoller class rolls either the die once or continuously, then as well as summarising the results.

### Code Formatting:
- **Is the code formatted using the Style Guidelines correctly?**
The code mostly adheres to style guidelines, but some variable names need adjustment for consistency. Overall the structure is great and includes key object oriented programming techniques.

### Readability:
- **Is the code easy to read?**
Yes, the code is generally easy to read. Proper indentation and comments enhance readability.

### Error Handling:
- **Are different errors handled correctly?**
Error handling is present, including validation of die roll values and sum calculations.

### Additional Suggestions:
- Refactor to ensure consistency in variable naming, especially across classes.

**Pull request merged**